File: Random/q1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        if (nums.length <= 1) {
            return null;
        }
        return twoSumRec(nums, target, 0);
        
    }
    
    private int[] twoSumRec(int[] nums, int target, int p) {
        if (p >= nums.length -1){
            return null;
        }
        for (int i = p + 1; i< nums.length; i++) {
            if (nums[i] + nums[p] == target) {
                int[] arr = {p,i};
                return arr;
            }
        }
        return twoSumRec(nums, target, p+1);
    }
}

File: Random/q1.cpp
#include <iostream>
#include<bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int,int> h;
        for(int i=0;i<n;i++){

            if(h.find(target-nums[i]) != h.end()){
                return {i,h[target-nums[i]]};
            }
            h.insert({nums[i],i});

        }
        return {};
    }
};


int main(int argc, char const *argv[])
{


    
    Solution a;
    vector<int> m =  {3,2,4};
    vector<int> ans = a.twoSum(m,6);

    for (int x: ans){
        cout<<x<<endl;
    }
    return 0;
}